Maple2.Model/Game/Event/GameEvent.cs

```diff
@@ -1,22 +1,146 @@
-﻿using Maple2.PacketLib.Tools;
+﻿using System;
+using System.Linq;
+using Maple2.Model.Enum;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
 using Maple2.Tools;
 
 namespace Maple2.Model.Game.Event;
 
-public class GameEvent {
-    public int Id { get; init; }
-    public string Name { get; init; }
-    public long BeginTime { get; init; }
-    public long EndTime { get; init; }
-    public GameEventInfo EventInfo { get; init; }
-}
+public class GameEvent : IByteSerializable {
+    public GameEventMetadata Metadata { get; init; }
+    public int Id => Metadata.Id;
+    public string Name => Metadata.Type.ToString();
+    public long StartTime => Metadata.StartTime.ToUnixTimeSeconds();
+    public long EndTime => Metadata.EndTime.ToUnixTimeSeconds();
+
+    public GameEvent(GameEventMetadata metadata) {
+        Metadata = metadata;
+    }
+
+    public bool IsActive() {
+        if (Metadata.StartTime > DateTimeOffset.UtcNow) {
```

Comment:
can we use a single variable assigned to `DateTimeOffset.UtcNow` rather than evaluating it multiple times. Feels easier to reason about the logic if you can guarantee the time you are checking against will not be changing within the function.